spec/requests/api/v1/cities/create_spec.rb

```diff
@@ -0,0 +1,59 @@
+describe 'POST /api/v1/countries/:country_id/cities', type: :request do
+  subject do
+    post api_v1_country_cities_path(country_id: country.id), headers: headers, params: params,
+                                                             as: :json
+  end
+
+  let(:headers) { auth_headers }
+  let(:user) { create(:user) }
+  let(:country) { create(:country) }
+  let(:city_name) { Faker::Address.city }
+  let(:params) do
+    {
+      city: {
+        name: city_name
+      }
+    }
+  end
+
+  context 'when response is successful' do
+    it_behaves_like 'a successful request'
+
+    it 'returns city info' do
+      subject
+
+      expect(json).to include_json(
+        country: {
+          id: country.id,
+          name: country.name
+        },
+        name: city_name
+      )
+    end
+
+    it 'creates the city' do
+      expect { subject }.to change { City.count }.by(1)
+    end
+  end
+
+  context 'when response is an error' do
+    context 'when request format is not valid' do
+      let(:params) { { names: city_name } }
+
+      it_behaves_like 'a bad request'
+    end
+
+    context 'when city name is already taken' do
+      let(:city) { create(:city, country: country) }
+      let(:city_name) { city.name }
+
+      it_behaves_like 'a bad request'
```

Comment:
When the server is able to understand the request body but it fails due to some validations, we should return [422 - Unprocessable Entity](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) instead of 400 - Bad Request.

BTW I don't see any `rescue` for `ActiveRecord::RecordInvalid` (this is the exception raised by ActiveRecord when you use `create!`, `update!` or `save!` and there is some validation error) https://github.com/szcrubyroostrap/rs-blackmarket/blob/development/app/controllers/api/concerns/errors_handling.rb